client/test/accounts_lib.spec.js

```diff
@@ -0,0 +1,95 @@
+'use strict'
+
+const assert = require('chai').assert
+
+const AccountsLib = require('../app/libs/accounts')
+
+
+describe("Accounts libs spec", () => {
+
+    describe("Validate and returns a new state", () => {
+
+        it("should returns a simple value in a new state", () => {
+            let state = {}
+            let nextState = AccountsLib.validateState('login', 'mail@cozy.io', state)
+
+            assert.isObject(nextState)
+            assert.notEqual(state, nextState)
+            assert.property(nextState, 'login')
+            assert.propertyVal(nextState, 'login', 'mail@cozy.io')
+        })
+
+        it("should override value if value already present in state", () => {
+            let state = {login: 'foo@gmail.com'}
+            let nextState = AccountsLib.validateState('login', 'mail@cozy.io', state)
+
+            assert.propertyVal(nextState, 'login', 'mail@cozy.io')
+        })
+
+        it("should ensure that `port` value is always a number", () => {
+            let nextState = AccountsLib.validateState('port', 143, {})
+            assert.isNumber(nextState.port)
+
+            nextState = AccountsLib.validateState('port', '993', {})
+            assert.isNumber(nextState.port)
+        })
+
+        it("should update imap / smtp login properties too when they're not custom", () => {
+            let nextState = AccountsLib.validateState('login', 'mail@cozy.io', {})
+
+            assert.property(nextState, 'imapLogin')
+            assert.property(nextState, 'smtpLogin')
+            assert.propertyVal(nextState, 'imapLogin', nextState.login)
+            assert.propertyVal(nextState, 'smtpLogin', nextState.login)
+
+            nextState = AccountsLib.validateState('login', 'box@cozy.io', nextState)
+
+            assert.propertyVal(nextState, 'imapLogin', nextState.login)
+            assert.propertyVal(nextState, 'smtpLogin', nextState.login)
+        })
+
+        it("should leave imap / smtp login properties untouched when they differ from login", () => {
```

Comment:
@m4dz : shouldnt you use a new group such as: `imap / smtp properties` for more lisibility?